### PR TITLE
Place Hid event on Uberfire

### PR DESCRIPTION
**Pull request by ederign**
When a editor is hid in ListBarWidgetImpl, we would like to generate a event (PlaceHidEvent). This should work like Focus/Maximized events.

We have to change ActivityManagerImpl, because when the part is associated with PathPlaceRequest, the place identifier was null (making impossible to
identify which place was hidden).

So in order to do that, when an activity is resolved by ActivityManagerImpl, I've made a lookup on resolvePlaceIdentifierForPathPlaceRequest method.

This feature request was generated by the need for hide the docks when Data Modeller was put in the background.

### CODE REVIEW

**Comment by manstis** [comment]
License headers and tests please!

**Comment by manstis** [comment]
Just a thought whether it'd be better to have PathPlaceRequest use IOC to get a reference to ActivityManager and resolve an identifier in the constructor (like we do for ObservablePath)? Otherwise the identifier is not set until a request to go to a PathPlaceRequest is handled by PlaceManager (and consequentially ActivityManager). I just think there's a greater risk of something else relying upon a PathPlaceRequest identifier in the future that doesn't pass through ActivityManager... IDK what ;-)

**Comment by ederign** [comment]
@manstis the news classes has license headers (PlaceHidEvent). Should I miss something?

About testing, I did not write tests for this new feature because ActivityManagerImpl nas no  unit test on it :( . 

I think we should cover everything in another task what do you think? (I can put that in my task list).

**Comment by manstis** [comment]
Hi Eder... I deleted my comment about missing license headers (forgetting I
was just looking at diffs; doh!) As for tests.... well I never thought I'd
hear you say that!!! Shocker! We should probably try to write something at
some time.

**Comment by ederign** [comment]
@manstis I fell bad about it, but it's a product bug. With pain in my heart I accept to leave the tests for later.



https://www.google.com.br/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&ved=0CAcQjRxqFQoTCOz1wdyX3scCFZGyHgodPDII3g&url=http%3A%2F%2Fwww.buzzfeed.com%2Fchelseamarshall%2Fpugs-feeling-so-left-out&psig=AFQjCNGi_LwkKE1y4rPsUqLdtUWMDRCDLw&ust=1441483787608271

File: uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/PanelManagerImpl.java [hunk 1]
```diff
@@ -290,6 +286,11 @@ public void onPartMinimized( final PartDefinition part ) {
     }
 
     @Override
+    public void onPartHid( final PartDefinition part ) {
+        placeHidEvent.fire( new PlaceHidEvent( part.getPlace() ) );
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/PanelManagerImpl.java]: @wmedvede [2]

File: uberfire-workbench/uberfire-workbench-client-views-bs2/src/main/java/org/uberfire/client/views/bs2/listbar/ListBarWidgetImpl.java [hunk 2]
```diff
@@ -369,6 +369,7 @@ public boolean selectPart( final PartDefinition part ) {
                 return true;
             }
             parts.add( currentPart.getK1() );
+            panelManager.onPartHid(currentPart.getK1());
```
Comment (ederign) [inline uberfire-workbench/uberfire-workbench-client-views-bs2/src/main/java/org/uberfire/client/views/bs2/listbar/ListBarWidgetImpl.java]: @wmedvede [1]

**Comment by manstis** [comment]
*uberfire*
(master) http://github.com/uberfire/uberfire/commit/fc41dbe45

(0.7.x) http://github.com/uberfire/uberfire/commit/3a63755fc

**Comment by manstis** [comment]
Thanks @ederign - BTW, I removed the ```ActivityManagerImplTest``` class (see comment above).

**Pull request closed without merging**